### PR TITLE
Updated Spec with Labels API

### PR DESCRIPTION
**Pull request by jenjenp**
Added the following:
List existing labels
Create new label
Add / Remove label

### CODE REVIEW

**Comment by jenjenp** [comment]
Added spec for moving events to inbox/trash/resolved - OO-1638

File: services.yaml [hunk 1]
```diff
@@ -1267,6 +1267,62 @@ paths:
...
+          name: service_id
+          description: Environment to use
+          type: string
+          required: true
+        - in: path
+          name: label_name
+          description: Label Name
+          type: string
+          required: true
+    put:
```
Comment (chook) [inline services.yaml:1305]: put should only be used if there's a corresponding get on that URL.
In this case we should use POST.
We may also want to drop the /{label_name} altogether as we provide it in the JSON

File: services.yaml [hunk 2]
```diff
@@ -1292,7 +1348,126 @@ paths:
...
+      summary: "Mark resolved"
+      description: "Marks the specified event as resolved"
+      responses:
+        200:
+          description: OK
+        401:
+          $ref: "#/definitions/UnauthorizedError"
+        404:
+          description: NOT FOUND
+  /services/{service_id}/events/{event_id}/labels:
```
Comment (chook) [inline services.yaml:1424]: i would put these before the new resolve/delete/inbox as the last APIs speak about labels.

File: services.yaml [hunk 3]
```diff
@@ -1292,7 +1348,126 @@ paths:
...
+          type: string
+          required: true
+        - in: path
+          name: event_id
+          description: Event to use
+          type: string
+          required: true
+    post:
+      tags:
+        - Labels
```
Comment (chook) [inline services.yaml:1414]: that is not precisely true. the user is not aware of it being a label

**Pull request merged**